pennylane/devices/default_qubit.py

```diff
@@ -161,6 +161,14 @@ def _conditional_broastcast_expand(tape):
     return (tape,), null_postprocessing
 
 
+@qml.transform
+def no_counts(tape):
+    """Throws an error on counts measurements."""
+    if any(isinstance(mp, qml.measurements.CountsMP) for mp in tape.measurements):
+        raise NotImplementedError("The JAX-JIT interface doesn't support qml.counts.")
```

Comment:
Is interface being jax the only reason why one would not allow counts in a circuit?

Reply:
This transform is only added if we are using jax-jit.  Converting samples -> counts is not jit friendly. We had been doing this check in `jax_jit.py`, but we are bypassing that logic now, so we need to add it back in.